doc/http-backoff.md

```diff
@@ -0,0 +1,10 @@
+# HTTP retry and backoff logic
+
+When Ignition is fetching a resource over http(s), if the resource is unavailable Ignition will continually retry to fetch the resource with an exponential backoff between requests. Once enough time has passed, Ignition will cancel the attempt and fail.
+
+There are two timeout values that can be configured with an Ignition config:
+
+- `httpResponseHeaders`: this field controls how long Ignition will wait for the server to send initial response headers for a given request attempt. If there is an outstanding request for a resource for which there has been no response, it will be cancelled when this timeout is reached. Ignition will then wait for a period of time and then try again. A value of 0 indicates that there is no timeout, and the default is 10 seconds.
+- `httpTotal`: this field controls how long Ignition will spend in total attempting to fetch a resource. If the amount of time spent making requests and waiting between failed attempts exceeds this value, Ignition will give up on fetching the resource and fail. A value of 0 indicates that there is no timeout, and the default is 0 seconds.
```

Comment:
I think this duplicates quite a bit of information from https://github.com/coreos/ignition/blob/master/doc/configuration-v2_1-experimental.md. Shouldn't we just reference that for the defaults and special syntax? Or are we trying here to split user-docs and dev-docs?